src/components/TopBar/Annotations.js

```diff
@@ -48,6 +48,30 @@ export const Annotations = observer(({ store, annotationStore }) => {
     return () => document.removeEventListener('click', handleClick);
   }, []);
 
+  useEffect(() => {
+    let _unresolvedComments = 0;
+    let _comments = 0;
+
+    commentStore.comments.forEach(obj => {
+      _comments++;
+
+      if (!obj.isResolved) _unresolvedComments++;
+    });
+
+    commentStore.annotation.setUnresolvedCommentCount(_unresolvedComments);
+    commentStore.annotation.setCommentCount(_comments);
+  }, [JSON.stringify(commentStore)]);
```

Comment:
I would propose observing each individual property that needs to be tracked instead of running heavy `JSON.stringify`

you can use the `observe()` function imported from mobx for this purpose

it will be something like this:

```
useEffect(() => {
 const runOnPropertyChange = () => {
  /* whatever you want to do */
 }

 const disposers = [/* properties to observe */].map(prop => {
   return observe(commentStore, prop, runOnPropertyChange, true);
 })

 return () => {
  disposers.forEach(dispose => dispose());
 }
}, []);
```

another option is to try precomputing the number of resolved/unresolved comments in the memoized value and using it as a trigger for the `useEffect`:

```
const commentsNumber = useMemo(() => {
  let _unresolvedComments = 0;
  let _comments = 0;

  commentStore.comments.forEach(obj => {
    _comments++;

    if (!obj.isResolved) _unresolvedComments++;
  });

  return [_comments, _unresolvedComments];
}, [commentStore.comments]);

useEffect(() => {
  const [comments, unresolved] = commentsNumber;
  commentStore.annotation.setUnresolvedCommentCount(unresolved);
  commentStore.annotation.setCommentCount(comments);
}, [commentsNumber])
```

any other options instead of serializing the store are acceptable as well